src/vconsole/vconsole-setup.c

```diff
@@ -53,6 +54,41 @@ static bool is_vconsole(int fd) {
         return ioctl(fd, TIOCLINUX, data) >= 0;
 }
 
+static int toggle_utf8(int fd, bool utf8) {
+        int r;
+        struct termios tc = {};
+
+        r = ioctl(fd, KDSKBMODE, utf8 ? K_UNICODE : K_XLATE);
+        if (r < 0)
+                return log_warning_errno(errno, "Failed to %s UTF-8 kbdmode: %m", utf8 ? "enable" : "disable");
```

Comment:
Hm, previous code in enable_utf8() first checked whether current mode is K_XLATE, and did not change the mode if it was not, to avoid messing with X11. Shouldn't it stay this way?